### ISSUE TITLE
Fix or remove SSHT code dependency 

### ISSUE THREAD

**Issue by CosmoMatt** [astro-informatics/s2fft#151]
Recently, the wheels for ssht (and therefore so3) are failing to build which is causing issues for some users @jasonmcewen. Our dependence on these modules is purely for testing purposes (this is the same case for healpy), however we now have well tested so called 'base' transforms, i.e. the loopy implementations, which we can reliably switch out here for such regression tests.

What I suggest is we simply remove these dependencies and cut the tie with this older code. How does this sit with you?

**Comment by jasonmcewen** [comment]
As we discussed @CosmoMatt , perhaps we can have two sets of test.  By default these regression tests would not be run, and so a depdencency on `ssht` would not be necessary.  But it would be useful to keep the test to hand in case we want to run them sometimes.  

Btw, I understand the `ssht` install issue has now been fixed (this was due to a breaking conan update to version 2; for now we have pinned conan 1 for `ssht`).
